### PR TITLE
Issue/520

### PR DESCRIPTION
**Pull request by fjarrett**
Implements #520 

Now our users can select the roles they want to create affiliate accounts in the Migration area.

By default, all roles that contain users are automatically selected upon page load.

I've included a user count for each role for convenience, while also disabling those roles which contain no users, so they aren't even selectable.

screen shot 2015-07-15 at 7 19 03 pm

If a user unchecks all the roles, the button will not be clickable. Because that's just silly.

screen shot 2015-07-15 at 7 23 06 pm

I've also improved the end process a little by redirecting the user to the Affiliates screen with the "added successfully" message.

screen shot 2015-07-15 at 7 23 54 pm

### CODE REVIEW

**Comment by fjarrett** [comment]
@pippinsplugins Please review.

https://github.com/affiliatewp/AffiliateWP/pull/776/files?w=1

File: includes/admin/tools/class-migrate-users.php [hunk 1]
```diff
@@ -42,64 +47,113 @@ public function process( $step = 1, $part = '' ) {
...
+			array(
+				'number'     => 100,
+				'offset'     => ( $step - 1 ) * 100,
+				'orderby'    => 'ID',
+				'order'      => 'ASC',
+				'meta_query' => array(
+					array(
+						'key'     => 'wp_capabilities',
+						'value'   => sprintf( '"(%s)"', implode( '|', $this->roles ) ),
+						'compare' => 'REGEXP'
```
Comment (pippinsplugins) [inline includes/admin/tools/class-migrate-users.php]: I did NOT know you could do this!

**Comment by pippinsplugins** [comment]
Works beautifully.

File: includes/admin/tools/class-migrate-users.php (hunk 1 above)
Comment (fjarrett) [inline includes/admin/tools/class-migrate-users.php]: Regex query on serialized data FTW 😎

**Pull request merged**